test/libsolidity/semanticTests/viaYul/dirty_memory_dynamic_array.sol

```diff
@@ -0,0 +1,19 @@
+contract C {
+    function f() public pure returns (bool correct) {
+        uint8[] memory m = new uint8[](1);
+        m[0] = uint8(0x0101);
+        assembly {
+            mstore(m, 257)
```

Comment:
Wait, if `m` is a dynamic array, this shouldn't change `m[0]`, should it? In the other example it does because it's a static array.

Reply:
Yes, this changes the length and not the first element.